### PR TITLE
Chore related to option page's UI

### PR DESCRIPTION
**Pull request by SunriseFox**
This PR includes:
+ extracted nav component
+ on clicking close/finish button, close the popup dialog instead of closing the tab
+ close the vide o track when it should be

Please test:
+ mobile

### CODE REVIEW

File: src/extension/options-page/Welcome/index.tsx [hunk 1]
```diff
@@ -68,7 +68,7 @@ const WelcomeActions = {
         this.autoVerifyBio(user, prove)
     },
     onFinish(reason: 'quit' | 'done') {
-        window.close()
+        ;((this as unknown) as { props: RouteComponentProps }).props.history.replace('/')
```
Comment (Jack-Works) [inline src/extension/options-page/Welcome/index.tsx]: Why?

File: src/extension/options-page/Welcome/index.tsx [hunk 2]
```diff
@@ -148,6 +148,13 @@ export default withRouter(function _WelcomePortal(props: RouteComponentProps<{ i
     const provePost = useValueRef(ProvePostRef)
     const identifier = useValueRef(IdentifierRef)
 
+    const onFinish = WelcomeActions.onFinish.bind({ props })
+    const closeDialog = (e: React.MouseEvent) => {
```
Comment (Jack-Works) [inline src/extension/options-page/Welcome/index.tsx]: I don't think it's the right way to close a dialog. This dialog is by-design to be always open, if you want to close it, just change the router to `/`.

File: src/extension/options-page/Welcome/index.tsx [hunk 3]
```diff
@@ -158,14 +165,14 @@ export default withRouter(function _WelcomePortal(props: RouteComponentProps<{ i
     }, [props.location.search])
 
     return (
-        <ResponsiveDialog open>
+        <ResponsiveDialog open onClick={closeDialog}>
```
Comment (Jack-Works) [inline src/extension/options-page/Welcome/index.tsx]: do you mean `onClose`?

File: src/extension/options-page/Welcome/index.tsx (hunk 2 above)
Comment (Jack-Works) [reply src/extension/options-page/Welcome/index.tsx]: And we have a close button, why detect target source?

File: src/extension/options-page/Welcome/index.tsx (hunk 1 above)
Comment (Jack-Works) [reply src/extension/options-page/Welcome/index.tsx]: You can change the parameter of this function.

File: src/components/Welcomes/Navigation/Navigation.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    const classes = useStyles()
+    return (
+        <nav className={classes.nav}>
+            {back ? (
+                <Button onClick={back} disableFocusRipple disableRipple className={classes.navBackButton}>
+                    <ArrowBack className={classes.navBackIcon} />
+                    {geti18nString('back')}
+                </Button>
+            ) : (
+                <></>
```
Comment (Jack-Works) [inline src/components/Welcomes/Navigation/Navigation.tsx]: You can use `null` here >_>

File: src/extension/options-page/Welcome/index.tsx (hunk 1 above)
Comment (SunriseFox) [reply src/extension/options-page/Welcome/index.tsx]: Using `this` is quite common these days... maybe. So how to make typescript recognize `this` the graceful way?

File: src/extension/options-page/Welcome/index.tsx [hunk 5]
```diff
@@ -148,6 +153,8 @@ export default withRouter(function _WelcomePortal(props: RouteComponentProps<{ i
     const provePost = useValueRef(ProvePostRef)
     const identifier = useValueRef(IdentifierRef)
 
+    const onFinish = WelcomeActions.onFinish.bind({ props })
```
[the remark this answers is not in the document]
Comment (SunriseFox) [reply src/extension/options-page/Welcome/index.tsx]: I tried to replace all `window.close` with this method as the tab window cannot be closed when using firefox (and currently there's no way to detect whether the window.close option succeeds).

Maybe we could add another function argument to pass the `props` or something? In fact I don't quite know how to do it the React way... I was just trying to take the React context into the function like Vue does.

You may modify it as your wish.

**Comment by Jack-Works** [comment]
Rebased master

**Pull request merged**